en/engine/event/event-emit.md

```diff
@@ -0,0 +1,150 @@
+
+![bubble-event](bubble-event.png)
+
+As shown in the picture above, when we send the event `“foobar”` from node c, if both node a and b listen to the event `“foobar”`, the event will pass to node b and a from c. For example:
+
+```ts
+// In the component script of node c
+this.node.dispatchEvent( new Event.EventCustom('foobar', true) );
+```
+
+To stop the event delivery after node b intercepts the event, call the function `event.propagationStopped = true` to do this. Detailed methods are as follows:
+
+```ts
+// In the component script of node b
+this.node.on('foobar', (event: EventCustom) => {
+  event.propagationStopped = true;
+});
+```
+
+> **Note**: to dispatch a custom event, do not use `cc.Event` because it's an abstract class. Instead, use `cc.Event.EventCustom` to dispatch a custom event.
+
+## Event object
+
+In the call-back of the event listener, the developer will receive an event object event of the `cc.Event` type. `propagationStopped` is the standard API of cc.Event, other important API include:
+
+| API Name                 | Type             | Meaning             |
+| :-------------             | :----------            |   :----------        |
+| type           | String   | The event type (event name).                      |
+| target          | cc.Node | The original target that received the event.                      |
+| currentTarget          | cc.Node | The current object that received the event. The current target of the event during the bubbling phase may be different from the original target.                     |
+| getType      | Function   | Get the event type.                   |
+| propagationStopped   | Boolean   | Whether or not stop the bubbling phase. The parent node of the current target no longer receives the corresponding event.                     |
+| propagationImmediateStopped              | Boolean   | Whether or not stop passing the current event immediately. The current target no longer receives the event either.                      |
+| detail             | Function | The information of the custom event, which belongs to cc.Event.EventCustom.    |
+| setUserData             | Function | Set the information of the custom event, which belongs to cc.Event.EventCustom.    |
+| getUserData             | Function | Get the information of the custom event, which belongs to cc.Event.EventCustom.    |
+
+Please refer to the `cc.Event` and API files of its child category for a complete API list.
+
+## System built-in event
+
+Above are the general rules for listening to events and emitting events. __Cocos Creator__ has built in system events. You can refer to the following documents:
```

Comment:
You can refer to the following documents:
-->
Please refer to the following documents: